### PR TITLE
feat: MetricsEngine table route (part 1)

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

make `TableRouteValue` an enum to add the variant of MetricsEngine table route

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/common/meta/src/key/table_route.rs [hunk 1]
```diff
@@ -38,42 +38,66 @@ impl TableRouteKey {
...
 
 #[derive(Debug, PartialEq, Serialize, Deserialize, Clone)]
-pub struct TableRouteValue {
+pub enum TableRouteValue {
+    Mito(MitoTableRouteValue),
+    Metrics(MetricsTableRouteValue),
+}
+
+#[derive(Debug, PartialEq, Serialize, Deserialize, Clone)]
+pub struct MitoTableRouteValue {
```
Comment (WenyXu) [inline src/common/meta/src/key/table_route.rs:47]: `MitoTableRouteValue` looks not good. The file table also uses this type of route. cc @evenyag @waynexia

File: src/common/meta/src/key/table_route.rs [hunk 2]
```diff
@@ -38,42 +38,66 @@ impl TableRouteKey {
...
+        self.mito_table_route()
+            .region_routes
             .iter()
             .find(|route| route.region.id == region_id)
             .cloned()
     }
+
+    pub fn region_routes(&self) -> &Vec<RegionRoute> {
+        &self.mito_table_route().region_routes
+    }
```
Comment (WenyXu) [inline src/common/meta/src/key/table_route.rs:93]: It's better to add some comments.

File: src/common/meta/src/key/table_route.rs [hunk 3]
```diff
@@ -269,7 +293,7 @@ impl TableRouteManager {
     ) -> Result<Option<RegionDistribution>> {
         self.get(table_id)
             .await?
-            .map(|table_route| region_distribution(&table_route.into_inner().region_routes))
+            .map(|table_route| region_distribution(table_route.into_inner().region_routes()))
```
Comment (WenyXu) [inline src/common/meta/src/key/table_route.rs:296]:
```suggestion
            .map(|table_route| region_distribution(table_route.region_routes()))
```

File: src/common/meta/src/key/table_route.rs (hunk 1 above)
Comment (MichaelScofield) [reply src/common/meta/src/key/table_route.rs:47]: Maybe file engine table should have its own `FileTableRouteValue`.

Comment (waynexia) [reply src/common/meta/src/key/table_route.rs:48]: I suspect if it is necessary to make variants over engine type...

Comment (fengjiachun) [reply src/common/meta/src/key/table_route.rs:48]: I think the route value type should not be determined by the engine type. There should only be two types: real and virtual.

**Comment by MichaelScofield** [comment]
@fengjiachun @waynexia @waynexia PTAL

File: src/common/meta/src/key/table_route.rs [hunk 4]
```diff
@@ -38,42 +38,70 @@ impl TableRouteKey {
...
+            .region_routes
             .iter()
             .find(|route| route.region.id == region_id)
             .cloned()
     }
+
+    /// Gets the [RegionRoute]s of this table route.
+    ///
+    /// # Panics
+    /// The engine type of this table is not "`Mito`".
```
Comment (WenyXu) [inline src/common/meta/src/key/table_route.rs:94]:
```suggestion
    /// Gets the [RegionRoute]s of this [TableRouteValue::Physical].
    ///
    /// # Panics
    /// The route type is not the [TableRouteValue::Physical].
```

**Review by WenyXu — APPROVED** [review]
Rest LGTM

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**